formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/schema/ProtoBufSchemaGenerator.kt

```diff
@@ -192,7 +195,16 @@ public object ProtoBufSchemaGenerator {
                 throw IllegalArgumentException("Field number $number is repeated in the class with serial name ${messageDescriptor.serialName}")
             }
 
-            append(' ').append(fieldName).append(" = ").append(number).appendLine(';')
+            append(' ').append(fieldName).append(" = ").append(number)
+
+            val isPackRequested = annotations.filterIsInstance<ProtoPacked>().singleOrNull() != null
+
+            when {
+                !isPackRequested -> appendLine(';')
+                !isList -> throw IllegalArgumentException("ProtoPacked annotation provided for ${messageDescriptor.getElementName(index)}: $fieldDescriptor, but packing is only valid on repeated fields (lists)")
```

Comment:
The actual format doesn't throw an exception, but rather silently ignores it. I'm not sure that there is a way to get the plugin to use dynamic logic to determine whether the annotation is valid (as it would also depend on the content of the list).

As the generation of the schema is more something to be done at development time the exception is here. However if you say that this is something that would be used at runtime (where a developer doesn't first hit the exception) I'd be happy to remove the checks (and ignore the annotation when generating the format). Perhaps the way to go would be to have a way to have serialization warnings.